src/main/java/uk/gov/hmcts/ccd/data/user/DefaultUserRepository.java

```diff
@@ -92,17 +95,22 @@ public UserDefault getUserDefaultSettings(final String userId) {
             LOG.debug("retrieving default user settings for user {}", userId);
             final HttpEntity requestEntity = new HttpEntity(securityUtils.authorizationHeaders());
             final Map<String, String> queryParams = new HashMap<>();
-            queryParams.put("uid", userId);
-            return restTemplate.exchange(applicationParams.userDefaultSettingsURL(),
-                HttpMethod.GET, requestEntity, UserDefault.class, queryParams).getBody();
-        } catch (HttpStatusCodeException e) {
+            queryParams.put("uid", ApplicationParams.encode(userId));
+            final String encodedUrl = UriComponentsBuilder.fromHttpUrl(applicationParams.userDefaultSettingsURL())
+                .buildAndExpand(queryParams).toUriString();
+            return restTemplate.exchange(new URI(encodedUrl), HttpMethod.GET, requestEntity, UserDefault.class)
+                .getBody();
+        } catch (RestClientResponseException e) {
             LOG.error("Failed to retrieve user profile", e);
-            final List<String> headerMessages = e.getResponseHeaders().get("Message");
+            final List<String> headerMessages = Optional.ofNullable(e.getResponseHeaders())
```

Comment:
@Daniel-Lam can we please verify that in case a user does not exist, we don't throw an error? We should instead just return 404 non found
Can we add a test for this case if not there yet please?

Reply:
@mario-paniccia - current behaviour is that User Profile API returns a `400 Bad Request` when the user does not exist; Data Store takes this response and throws a `BadRequestException` with that `400` error. Do we want to change this behaviour, then?

Reply:
If so, which side should change: Data Store or User Profile API?

Reply:
Response from @mario-paniccia on Slack earlier today:
> Mario 7:51 AM
> morning Daniel. I'm thinking better leave it as it is now. Because I've realised it might somehow impact the UI as well if we change this now. So it's quite a big work. Not worth it. Cheers though